### PR TITLE
v4l2: Add enumerate_frameintervals()

### PR DESCRIPTION
**Pull request by DmitrySamoylov**
Hey! 

In this PR I've inplemented this ioctl function:
https://www.kernel.org/doc/html/v4.10/media/uapi/v4l/vidioc-enum-frameintervals.html

### CODE REVIEW

File: examples/frameintervals.rs [hunk 1]
```diff
@@ -0,0 +1,44 @@
+extern crate clap;
+extern crate v4l;
+
+use clap::{App, Arg};
+use v4l::CaptureDevice;
+
+fn main() {
+    let matches = App::new("v4l device")
+        .version("0.2")
+        .author("Christopher N. Hesse <raymanfx@gmail.com>")
```
Comment (raymanfx) [inline examples/frameintervals.rs:10]: This should have your name on it.

File: src/capture/device.rs [hunk 2]
```diff
@@ -1,10 +1,11 @@
+use std::convert::TryFrom;
 use std::{io, mem, path::Path};
 
 use crate::capture::{Format, Parameters};
 use crate::device;
 use crate::v4l2;
 use crate::v4l_sys::*;
-use crate::{DeviceInfo, FormatDescription, FourCC, Fraction};
+use crate::{DeviceInfo, FormatDescription, FourCC, Fraction, FrameInterval};
```
Comment (raymanfx) [inline src/capture/device.rs:8]: Won't this conflict with your other change? Maybe rebase this one on top of the other one so I can merge them after one another.

Comment (DmitrySamoylov) [reply src/capture/device.rs]: Done!

**Comment by raymanfx** [comment]
Okay, let's get this in.
Can you rebase this so it's only one commit and keep the other commit in the other PR? We're good to go then I think.

**Comment by DmitrySamoylov** [comment]
Hm, not sure what you mean. It is one commit per PR now. This branch is based on another branch from previous PR. So there are only two commits (and two branches)

**Comment by raymanfx** [comment]
You are right, I got confused there. Thank you for submitting the work!

**Pull request merged**